plugin/pkg/scheduler/factory/plugins.go

```diff
@@ -356,9 +359,23 @@ func getPriorityFunctionConfigs(names sets.String, args PluginFactoryArgs) ([]al
 			})
 		}
 	}
+	var totalPriority int
+	for _, config := range configs {
+		// Checks for overflow and returns errors.
+		// TODO: Need to check overflow for weight*MaxPriority.
```

Comment:
haha this simple PR is getting pretty interesting :).  Yeah, k82cn's idea is reasonable, if we can move this thing into a generic place (make sure to create corresponding unit test) thats great.